lib/kernel/src/application_controller.erl

```diff
@@ -1684,10 +1700,7 @@ get_env_key([H | T], Key, Res) ->
 get_env_key([], _Key, Res) -> Res.
 
 add_env(Name, Env) ->
-    foreach(fun({Key, Value}) ->
-			  ets:insert(ac_tab, {{env, Name, Key}, Value})
-		  end,
-		  Env).
+    ets:insert(ac_tab, [{{env, Name, Key}, Value} || {Key, Value} <- Env]).
```

Comment:
Not sure if this is relevant here but note that this no longer crashes on lists that contain something else than 2-tuples.

Reply:
Yup, I checked the call sites and they all validate the data before, so we are good!